src/command/process_update_task.rs

```diff
@@ -362,17 +588,21 @@ impl AsyncRunnable for ProcessUpdateTask {
                 return Ok(());
             }
         };
-
-        if let Err(error) = processor.process().await {
+        let result = processor.process().await;
+        if let Err(error) = result {
```

Comment:
if you need both cases (err, ok), you should use match

Reply:
[Matching here is better](https://github.com/pxp9/weather_bot_rust/pull/29/commits/ffe1a5329c27366b2b2240cdfdf82c3d40cbdab9)